app/models/miq_request.rb

```diff
@@ -429,7 +429,7 @@ def execute
       :instance_id    => id,
       :method_name    => "create_request_tasks",
       :zone           => options.fetch(:miq_zone, my_zone),
-      :role           => my_role,
+      :role           => kind_of?(ServiceTemplateProvisionRequest) ? 'automate' : my_role,
```

Comment:
The `my_role` method is used so sub-classes can override it, like you point out in the description for `AutomationRequest`.  This value should be changed in the `my_role` method in the `ServiceTemplateProvisionRequest` model here [app/models/service_template_provision_request.rb#L36-L38](https://github.com/ManageIQ/manageiq/blob/05e85b1f34fb1007cf2bc6de77844a8c62c1a749/app/models/service_template_provision_request.rb#L36-L38)

Then you do not need this change to the base `miq_request` model.

Reply:
@gmcculloug I specifically changed it that way because I want to limit the change to only the create_request_tasks call.

Reply:
It would be clearer if the `my_role` method took a parameter and the model could then determine the proper response.  The refactoring would effect several files so I am ok with doing that as a follow to this PR to this.

@tinaafitz @mkanoor Any thoughts on this approach?